apiary.apib

```diff
@@ -3181,6 +3183,7 @@ Get a list of project milestones.
                     + closed
             + due_before: `2017-01-01` (string, optional)
             + due_after: `2017-01-01` (string, optional)
+            + term: `coffee` (string, optional) - Searches for a term in title
```

Comment:
If it's only searching on the title let's call the filter `title`. The `term` filter on contacts/companies searches on more than one property eg. name, phone number, company, ...

```suggestion
            + title: `coffee` (string, optional) - Searches for a term in title
```

Reply:
Hm, but if we decide to also search in the description (or other fields, like the project title/description) in one month, we will need to rename this to 'term', then, causing a breaking change/API version change.

Besides that, as an API consumer, I find it weird/inconsistent to have some endpoints use `term` as keyword search field and other endpoints in the same API use another field for keyword search. The slight difference that in one endpoint the search is done on two fields while in another endpoint the search is done in only one field (for now) is not enough to sustain such an inconsistency, in my point of view.

Reply:
Why not have both? Add a `title` filter now to only search on the title. And if we want a more general search term add `term`. No breaking changes needed.

Reply:
You can add a `term` filter if you want. Just make sure you keep in mind that integrators will use it to search on more than the title.

Reply:
Imo this depends on the current knowledge about which filters we're still going to add...
If searching on more than "title" is currently planned, it's OK to go for "term".
If nothing is planned, by yagni and call it "title".